testdata/script/reverse.txtar

```diff
@@ -119,25 +112,6 @@ func printStackTrace(w io.Writer) error {
 	_, err := w.Write(stack)
 	return err
 }
--- build-error/error.go --
-package p
-
-import "reflect"
-
-// This program is especially crafted to work with "go build",
-// but fail with "garble build".
-// This is because we attempt to convert from two different struct types,
-// since only the anonymous one has its field name obfuscated.
-// This is useful, because we test that build errors can be reversed,
-// and it also includes a field name.
```

Comment:
Hm, this error test case was useful for the reason explained here. You're right that garble is getting smart enough that the "intended failure" no longer triggers.

I think we should find a different kind of build error that still includes a position, a type name, and a field name. It doesn't matter if it also fails with `go build`, because we only use this file with `garble build`.

How about this one? https://go.dev/play/p/UjFrUaf7PcJ

Reply:
No we need something which passes with go build and not garble otherwise the output will be already not obfuscated.

Reply:
Right, I keep forgetting that garble build begins with an unobfuscated typecheck.

Then we need to figure out how to make this test case work again. Perhaps try putting the obfuscated struct in an imported package, so that the per-package reflect detection does not kick in. We might fix that at some point, but perhaps not, given the cost.

Reply:
That's one of the next issues I want to fix, to (hopefully) mitigate reflection issues once and for all. 

Sure we could construct some case, but a goal of garble is to be interchangeable with the go compiler, so I don't think that we should have a test case asserting the opposite.

Reply:
> Sure we could construct some case, but a goal of garble is to be interchangeable with the go compiler, so I don't think that we should have a test case asserting the opposite.

Fair point, it's a bit of a weird one. I think it's also pointless to try to test something that the user shouldn't really run into.